united.py

```diff
@@ -28,10 +54,224 @@ def parse_date(created):
     hour = time.strftime('%H')
-    hour = int(hour) if created.endswith('AM') else int(hour)+12
+    hour = int(hour) if created.endswith('AM') else int(hour) + 12
     return datetime(time.year, time.month, time.day, hour, time.minute, 0)
 
 
+def is_nth_weekday(nth, daynum, year, month):  # start counting the daynum from monday = 0
+    return calendar.Calendar(nth).monthdatescalendar(
+        year,
+        month
+    )[daynum][0]
+
+
+def get_parent_object_node(node):
+    while node.parentNode:
+        node = node.parentNode
+        if node.nodeName == "Object":
+            return node
+
+
+def all_station_in_date_frame(created):
+
+    station_data_in_date = collection.getElementsByTagName('date_selected')
+    station_data_in_date.sort(key=lambda x: int(x.attributes['Position'].value))
+    accident_date = parse_date(created)
+    summer_clock = is_nth_weekday(4, 4, accident_date[0], 3)
+    winter_clock = is_nth_weekday(4, 6, accident_date[0], 10)
+
+    # weather is given in UTC time
+    # therefore in summer clock we deduct 3 hours from the local time and in winter clock 2 hours
+    # [
+    # if accident happend between april and september
+    if accident_date.month < 10 & accident_date.month > 3:
+        accident_date.hour -= 3
+
+    # if accident happend before the last sunday of october at 2:00 o'clock
+    elif accident_date[1] == 10 & (
+                winter_clock[2] > accident_date[2] | (winter_clock[2] == accident_date[2] & accident_date[3] < 2)):
+        accident_date[3] -= 3
+
+    # if accident happend after the last friday of march at 2:00 o'clock
+    elif (accident_date[1] == 3 & summer_clock[2] < accident_date[2] | (
+                    summer_clock[2] == accident_date[2] & accident_date[3] >= 2)):
+        accident_date[3] -= 3
+    else:  # winter_clock
+        accident_date[3] -= 2
+    # ]
+    adate = ''.join((accident_date.year, accident_date.month, accident_date.day, accident_date.hour))
+
+    i = -1
+
+    doc = Document()
+    base = doc.createElement('accident_date')
+    doc.appendChild(base)
+
+    for station in station_data_in_date:
+        i += 1
+        if adate in str(station.childNodes[i].nodeValue):
+            base.appendChild(get_parent_object_node(station))
+    print doc.toprettyxml(indent="    ", encoding="utf-8")
+
+
+
+def find_station_by_coordinate(latitude, longitude):
+
+    station_place_in_xml = -1
+    lat_difference = 0
+    lon_difference = 0
+    min_distance = 100000  # initialize big starting value so the distance will always be smaller than the initial
+    station_data = collection.getElementsByTagName('surface_station')
+    i = -1
+
+    for station in station_data:
+        i += 1  # save the place of the station in the file
+        station_lon = station.getElementsByTagName('station_lon')
+        for lon in station_lon:
+            lon_difference = math.pow(float(lon.childNodes[0].nodeValue) - float(longitude), 2)
```

Comment:
no, it's supposed to be replaced each time , i'm checking for the closest
station therefore each iteration i'm getting a different longitude
difference, after that i'm doing the distance equation and if it's smaller
replace the current the min_distance value

2016-04-17 9:54 GMT+03:00 Daniel Hershcovich <notifications@github.com>:

> In united.py
> <https://github.com/hasadna/anyway/pull/579#discussion_r59979173>:
>
> > +
> > +
> > +def find_station_by_coordinate(latitude, longitude):
> > +
> > +    station_place_in_xml = -1
> > +    lat_difference = 0
> > +    lon_difference = 0
> > +    min_distance = 100000  # initialize big starting value so the distance will always be smaller than the initial
> > +    station_data = collection.getElementsByTagName('surface_station')
> > +    i = -1
> > +
> > +    for station in station_data:
> > +        i += 1  # save the place of the station in the file
> > +        station_lon = station.getElementsByTagName('station_lon')
> > +        for lon in station_lon:
> > +            lon_difference = math.pow(float(lon.childNodes[0].nodeValue) - float(longitude), 2)
>
> lon_difference is replaced each time. Did you intend to use +=?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hasadna/anyway/pull/579/files/2dd7573bd5bab0315ca68dc10db70ec63fcc8e09#r59979173>
>